src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1700,28 +1701,30 @@ private void CalculateIORedirection(bool isWindowsApplication, out bool redirect
             "Performance",
             "CA1822:Mark members as static",
             Justification = "Accesses instance members in preprocessor branch.")]
+        [SkipLocalsInit]
         private bool IsExecutable(string path)
         {
 #if UNIX
             return Platform.NonWindowsIsExecutable(this.Path);
 #else
 
             string myExtension = System.IO.Path.GetExtension(path);
+            string pathext = Environment.GetEnvironmentVariable("PATHEXT");
 
-            var pathext = Environment.GetEnvironmentVariable("PATHEXT");
-            string[] extensionList;
-            if (string.IsNullOrEmpty(pathext))
-            {
-                extensionList = new string[] { ".exe", ".com", ".bat", ".cmd" };
-            }
-            else
-            {
-                extensionList = pathext.Split(';');
-            }
+            ReadOnlySpan<char> extensionList = string.IsNullOrEmpty(pathext)
+                ? ".exe;.com;.bat;.cmd"
+                : pathext;
+
+            int extensionCount = extensionList.Count(';') + 1;
+            Span<Range> extensions = extensionCount < 0x20
+                ? (stackalloc Range[0x20]).Slice(0, extensionCount)
+                : new Range[extensionCount];
 
-            foreach (string extension in extensionList)
+            extensions.Clear();
+            extensions = extensions.Slice(0, extensionList.Split(extensions, ';'));
```

Comment:
Do we really need the Clear(). I thing .Net does this for us.

Reply:
I did actually mean to remove the `.Clear()` here because I think `Split` will be writing in a way where it doesn't matter if the stack buffer is zeroed.

That said, it's worth noting that I included it due to adding the `SkipLocalsInit` decoration. It's a common recommendation when using stack allocated buffers because .NET's default zeroing of stack allocated memory is very slow according to the .NET folks.